omod-common/src/main/java/org/openmrs/module/webservices/rest/web/filter/AuthorizationFilter.java

```diff
@@ -70,6 +70,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 			HttpServletResponse httpresponse = (HttpServletResponse) response;
 			httpresponse.sendError(HttpServletResponse.SC_FORBIDDEN, "IP address '" + request.getRemoteAddr()
 			        + "' is not authorized");
```

Comment:
Did you intentionally make changes to this class?

Reply:
Yes. I added the return statement because I found that deserialization was occurring even after an IP failed the "allowed IPs" check. I figured it makes sense to stop processing a request's payload after it fails the "allowed IPs" authorization check, but this y isn't strictly necessary to patch the vulnerability. If you think it's going to cause instability somewhere else in the system I can take it out.

Reply:
I have not deeply evaluated the effect of this. Just saw this comment on line 104 "continue with the filter chain in all circumstances". Does it matter if you just did chain.doFilter(request, response); before returning?

Reply:
It does, unfortunately: a lot of processing happens in the `chain.doFilter(request, response)` call tree, to include deserialization. The purpose of the return statement is to prevent all that and just go straight to sending an HTTP 403 (Forbidden) response when appropriate.

Reply:
Fair enough. 😊